database/database_handler.py

```diff
@@ -0,0 +1,88 @@
+import os
+import subprocess
+import json
+
+from sqlalchemy import create_engine, Column, String, Boolean, Date, Integer, Float
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.exc import SQLAlchemyError
+
+from utils.general import LOGGER
+
+LOCAL_RUN = False
+
+
+def get_db_access_token(client_id):
+    # Authenticate using Managed Identity (MSI)
+    try:
+        command = ["az", "login", "--identity", "--username", client_id]
+        subprocess.check_call(command)
+    except subprocess.CalledProcessError as e:
+        print("Error during 'az login --identity':", e)
+        raise e
+
+    # Execute Azure CLI command to get the access token
+    command = ["az", "account", "get-access-token", "--resource-type", "oss-rdbms"]
+    output = subprocess.check_output(command)
+
+    # Parse the output to retrieve the access token
+    access_token = json.loads(output)["accessToken"]
+
+    return access_token
+
+
+def make_connection_string():
```

Comment:
In a Class setup this can become a private method. Private methods don't necessarily exists in Python like other languages but they can be imitated adding a `_` in front of the name: `_make_connection_string`.
This basically allows to who uses your class know that this is not a function they should use, but they should use `create_connection`.